src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,246 @@
+
+  // Local properties
+  amalgamatingBusinessesValid = false
+  amalgamatingBuinesses = []
+  initialBusinessLookupObject = EmptyBusinessLookup
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  async setAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
+
+    // Get the address of the amalgamating business
+    if (businessLookup.identifier && business.foundingDate) {
+      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
+        .then((data) => {
+          // SP and GP have businessOffice instead of registeredOffice
+          return data?.registeredOffice || data?.businessOffice
+        }).catch(() => {
+          return undefined
+        })
+      if (addresses) {
+        business.officeAddress = addresses
+      }
+    }
+
+    // If the amalgamating businesses array is not empty, check if identifier already exists.
+    // If identifier already exists, don't add the business to the array.
+    if (this.amalgamatingBuinesses.length > 0) {
+      const filteredBusinesses = this.amalgamatingBuinesses.filter(function (id) {
+        return id.identifier === business.identifier
+      })
+      if (filteredBusinesses.length === 0) {
+        this.amalgamatingBuinesses.push(business)
+      }
+    } else {
+      this.amalgamatingBuinesses.push(business)
+    }
```

Comment:
Prevent from adding a business that's already been added to the array.

Reply:
Use `find` instead of `filter`. (Find returns the first one. Filter does more work and checks all entries.)